### PR TITLE
Allow a configurable prefix to prepend the metric

### PR DESCRIPTION
**Pull request by objectuser**
_Note: I have an alternative approach in another PR coming._

I added a `Modifiers` module to house the prefix. This could be enhanced with other modifiers or merged into the `Context` module.

```
config :metrix,
  metric_prefix: "my-prefix."
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,20 @@ config :metrix,
   context: %{"source" => "my-app"}
 ```
 
+An optional (global) prefix may also be configured:
+
+```elixir
+config :metrix,
+  metric_prefix: "my-prefix."
```
Comment (rwdaigle) [inline README.md:190]: Can we remove the `metric_` part of the name here (and throughout) to be consistent w/ how we're handling the global `context`?

File: lib/metrix.ex [hunk 2]
```diff
@@ -22,6 +24,18 @@ defmodule Metrix do
     end
   end
 
+  defp initial_modifiers do
```
Comment (rwdaigle) [inline lib/metrix.ex:27]: What's the purpose of this pass-through function? Seems like we only need one function to get the initial modifiers.

File: lib/modifiers.ex [hunk 3]
```diff
@@ -0,0 +1,38 @@
+defmodule Metrix.Modifiers do
```
Comment (rwdaigle) [inline lib/modifiers.ex:1]: I like the idea of having a single Agent handle all global state (context and prefix). If you want to do that, I'd be 👍 

Wonder if it should be a separate PR so we don't conflate things?

File: README.md (hunk 1 above)
Comment (objectuser) [reply README.md:190]: Yes, it was only for clarity wrt what is being prefixed.

File: lib/metrix.ex (hunk 2 above)
Comment (objectuser) [reply lib/metrix.ex:27]: You're right, I was thinking ahead and got lost.

File: lib/modifiers.ex (hunk 3 above)
Comment (objectuser) [reply lib/modifiers.ex:1]: Makes sense to me.

File: lib/metrix.ex [hunk 4]
```diff
@@ -22,6 +24,18 @@ defmodule Metrix do
...
+    %{}
+    |> metric_prefix_modifier
+  end
+
+  defp metric_prefix_modifier(modifiers) do
+    case Application.get_env(:metrix, :metric_prefix) do
```
Comment (objectuser) [inline lib/metrix.ex]: Thinking that having the `:metric_prefix` (or, now, `:prefix`) atom in both modules is a bit of a code smell.
